src/codec.rs

```diff
@@ -1099,13 +1100,36 @@ impl Decode for bool {
 	}
 }
 
+impl Encode for Duration {
+	fn size_hint(&self) -> usize {
+		mem::size_of::<u64>() + mem::size_of::<u32>()
+	}
+
+	fn encode(&self) -> Vec<u8> {
+		let secs = self.as_secs();
+		let nanos = self.subsec_nanos();
+		(secs, nanos).encode()
+	}
+}
+
+impl Decode for Duration {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		let (secs, nanos) = <(u64, u32)>::decode(input)?;
+		if nanos >= A_BILLION {
+			Err("Number of nanoseconds should not be higher than 10^9.".into())
+		} else {
+			Ok(Duration::new(secs, nanos))
+		}
+	}
+}
```

Comment:
oups I forget we have to implement encodelike itself.

```suggestion
}

impl EncodeLike for Duration {}
```